lib/ipni-client.js

```diff
@@ -10,22 +10,24 @@ import { decodeBase64, decodeVarint } from '../vendor/deno-deps.js'
  * }>}
  */
 export async function queryTheIndex (cid, providerId) {
-  const url = `https://cid.contact/cid/${encodeURIComponent(cid)}`
-
   let providerResults
   try {
-    const res = await fetch(url)
-    if (!res.ok) {
-      console.error('IPNI query failed, HTTP response: %s %s', res.status, (await res.text()).trimEnd())
-      return { indexerResult: `ERROR_${res.status}` }
-    }
-
-    const result = await res.json()
-    providerResults = result.MultihashResults.flatMap(r => r.ProviderResults)
+    providerResults = await pRetry(() => getRetrievalProviders(cid), {
+      retries: 5,
+      shouldRetry: (error) => {
+        return error.statusCode && error.statusCode >= 500
+      },
+      onFailedAttempt: (error) => {
+        console.error(error)
+        console.error('IPNI query failed, retrying...')
+      }
+    })
     console.log('IPNI returned %s provider results', providerResults.length)
   } catch (err) {
     console.error('IPNI query failed.', err)
-    return { indexerResult: 'ERROR_FETCH' }
+    return {
+      indexerResult: typeof err.statusCode === 'number' ? `ERROR_${err.statusCode}` : 'ERROR_FETCH'
```

Comment:
```suggestion
      indexerResult: typeof err.status === 'number' ? `ERROR_${err.status}` : 'ERROR_FETCH'
```

https://developer.mozilla.org/en-US/docs/Web/API/Response/status

Reply:
See also above

Reply:
Oh I see, this is the error from `assertOkResponse`, not `fetch`. Nevermind